SD-K9/src/app/providers/svg-manager.service.spec.ts

```diff
@@ -1,26 +1,211 @@
 import { TestBed } from '@angular/core/testing';
 
-import { HttpClientModule } from '@angular/common/http';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
 import { SVGManager } from './svg-manager.service';
+import { SVGCoordinate } from '../models/svg-coordinate.model';
 
 describe('SVGManager', () => {
     let service: SVGManager;
+    let httpMock: HttpTestingController;
+
+    const dummySVG = `<svg version="1.1" x="0px" y="0px" viewBox="0 0 10 10">
+    <g id="nodes">
+    <circle class="node" cx="8" cy="2" r="0.5"/>
+    <circle class="node" cx="2" cy="8" r="0.5"/>
+    </g>
+    <g id="classes">
+    <circle class="classroom" id="A" cx="2" cy="2" r="0.5"/>
+    <circle class="classroom" id="B" cx="8" cy="8" r="0.5"/>
+    </g>
+    <g id="stairs">
+    <circle id="stair-1" cx="5" cy="5" r="0.5"/>
+    <circle id="stair-2" cx="5" cy="2" r="0.5"/>
+    </g>
+    <g id="escalators">
+    <g id="up">
+    <circle id="up-escalator-1" cx="5" cy="5" r="0.5"/>
+    <circle id="up-escalator-2" cx="8" cy="8" r="0.5"/>
+    </g>
+    </g>
+    </svg>`;
 
     beforeEach(() => {
         TestBed.configureTestingModule({
-            imports: [HttpClientModule],
+            imports: [HttpClientTestingModule],
             providers: [
                 SVGManager
             ]
         }).compileComponents();
         service = TestBed.get(SVGManager);
+        httpMock = TestBed.get(HttpTestingController);
+        spyOn(service, 'drawSVGPath').and.callThrough();
     });
 
     afterEach(() => {
         service = null;
+        httpMock.verify();
     });
 
     it('should be created', () => {
         expect(service).toBeTruthy();
     });
-});
\ No newline at end of file
+
+    it('#getSVG should get SVG from assets', () => {
+
+        service.getSVG('any/0').subscribe(floorplan => {
+            expect(floorplan).toEqual(dummySVG);
+        });
+
+        const request = httpMock.expectOne('../../assets/any/0.svg');
+        expect(request.request.method).toBe('GET');
+
+        request.flush(dummySVG);
+    });
+
+    it('#getWalkableNodes should return a list of walkable nodes from the svg', () => {
+        const building = 'any';
+        const floor = 0;
+
+        const expectedNodes = [
+            { id: '', building, floor, x: 8, y: 2 },
+            { id: '', building, floor, x: 2, y: 8 }
+        ];
+
+        service.getWalkableNodes(building, floor)
+            .then(nodes => {
+                expect(nodes.length).toEqual(2);
+                expect(nodes).toEqual(expectedNodes);
+            })
+            .catch((error) => {
+                fail(error);
+            });
+
+        const request = httpMock.expectOne('../../assets/any/0.svg');
+        expect(request.request.method).toBe('GET');
+
+        request.flush(dummySVG);
+    });
+
+    it('#getClassroom should return a classroom given an id', () => {
```

Comment:
It also requires building and floor:
```suggestion
    it('#getClassroom should return a classroom given an id, building and floor', () => {
```